tests/js/server/aql/aql-traversal-paths-optimization.js

```diff
@@ -0,0 +1,216 @@
+        // PRUNE
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p.edges", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.vertices[-1].testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.vertices[-1].testi == 'abc' RETURN p.edges", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.edges[-1].testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.edges[-1].testi == 'abc' RETURN p.edges", [ false, true, false ]],
+
+        // weighted traversals
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p", [ true, true, true ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p[0]", [ true, true, true ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p.weights", [ false, false, true ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p.edges", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN [p.edges, p.vertices]", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN [p.edges, p.weights]", [ false, true, true ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' OPTIONS { order: 'weighted', weightAttribute: 'testi' } RETURN p.testi", [ true, false, false ]],
+
+      ].forEach((query) => {
+        let nodes = AQL_EXPLAIN(query[0]).plan.nodes;
+        let traversal = nodes.filter((node) => node.type === 'TraversalNode')[0];
+
+        assertEqual("p", traversal.pathOutVariable.name);
+
+        assertEqual(query[1][0], traversal.options.producePathsVertices);
+        assertEqual(query[1][1], traversal.options.producePathsEdges);
+        assertEqual(query[1][2], traversal.options.producePathsWeights);
```

Comment:
This is good test-suite.

I would like to make sure that the result does not contain the other values.

I see two options here.
1) AFAIR we have "UnitTests" for TraversalBlocks. There we can inject the expected produce setting and ensure that the resulting AQL value only has the asked for Parts. I think that is the easiest way to implement this.
2) We could take a small sample of the Above values, run them and take a look at the "scannedIndex" value in the resulting Cursor getExtra(). for all queries that do not produce vertices the value should be lower as we can avoid reading those documents all together. We cannot assert weight or edges this way.
3) Maybe? AFAIR we had a way to "create" a plan, expose it as JSON, and execute a JSON plan as AQL. We could take a simple `RETURN p` query (the optimization is NOT applied) then we "plan" it. And overwrite the .produceXXX values in the plan. Then we regenerate the query from the Plan => The returned p should only contain what we asked for. I am not 100% sure if this works.